File: library/src/main/java/name/zeno/android/system/ZClipboard.java
package name.zeno.android.system;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;

import name.zeno.android.util.ZLog;

/**
 * 剪贴板
 *
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/13.
 */
public class ZClipboard
{
  public static void setText(Context context, String txt)
  {
    setText(context, null, txt);
  }

  // 设置剪贴板内容
  public static void setText(Context context, String label, String txt)
  {
    ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
    clipboard.setPrimaryClip(ClipData.newPlainText(label, txt));
  }

  // 获取剪贴板内容
  public static String[] getText(Context context)
  {
    String           result    = null;
    String           label     = null;
    ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
    ClipData         data      = clipboard.getPrimaryClip();
    if (data != null && data.getItemCount() > 0 && data.getItemAt(0).getText() != null) {
      result = data.getItemAt(0).getText().toString();
      CharSequence l = data.getDescription().getLabel();
      label = l == null ? null : l.toString();
    }

    return new String[]{label, result};
  }
}

File: library/src/main/java/name/zeno/android/system/ZClipboard.kt
package name.zeno.android.system

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context

/**
 * 剪贴板
 *
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/13.
 */
object ZClipboard {
  // 设置剪贴板内容
  fun setText(context: Context, clipboard: Clipboard?) {
    val cm = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    cm.primaryClip = ClipData.newPlainText(clipboard?.label, clipboard?.txt)
  }

  /** # 获取剪贴板内容 */
  fun getText(context: Context): Clipboard? {
    var result: String? = null
    var label: String? = null
    val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    val data = clipboard.primaryClip
    if (data != null && data.itemCount > 0 && data.getItemAt(0).text != null) {
      result = data.getItemAt(0).text.toString()
      val l = data.description.label
      label = l?.toString()
    }

    return when {
      result != null -> Clipboard(result, label)
      else -> null
    }
  }
}
